File: Templates/TypeEditor/src/TemplateSolutions/VB.CustomTypeEditor/CustomControlLibrary.Server/TypeRoutingProvider.vb
Imports Microsoft.DotNet.DesignTools.TypeRouting

Namespace CustomControlLibrary.Designer.Server
    ''' <summary>
    '''  Class holding the TypeRoutings for resolving the control designer type on the server.
    ''' </summary>
    <ExportTypeRoutingDefinitionProvider>
    Friend Class TypeRoutingProvider
        Inherits TypeRoutingDefinitionProvider

        Public Overrides Function GetDefinitions() As IEnumerable(Of TypeRoutingDefinition)
            Return {New TypeRoutingDefinition(TypeRoutingKinds.Designer, NameOf(CustomControlDesigner), GetType(CustomControlDesigner))}
        End Function
    End Class
End Namespace

File: Templates/TypeEditor/src/TemplateSolutions/CS.CustomTypeEditor/CustomControlLibrary.Server/TypeRoutingProvider.cs
using System.Collections.Generic;
using Microsoft.DotNet.DesignTools.TypeRouting;

namespace CustomControlLibrary.Designer.Server
{
    /// <summary>
    ///  Class holding the TypeRoutings for resolving the control designer type on the server.
    /// </summary>
    [ExportTypeRoutingDefinitionProvider]
    internal class TypeRoutingProvider : TypeRoutingDefinitionProvider
    {
        public override IEnumerable<TypeRoutingDefinition> GetDefinitions()
            => new[]
            {
                new TypeRoutingDefinition(
                    TypeRoutingKinds.Designer,
                    nameof(CustomControlDesigner),
                    typeof(CustomControlDesigner))
            };
    }
}
